src/foundry/client/pixi/sources/rendered-source.d.ts

```diff
@@ -16,28 +16,31 @@ declare global {
   }
 
   namespace RenderedPointSource {
-    type RenderedPointSourceAnimationConfig = {
+    type RenderedPointSourceAnimationConfig<AnimationValue> = InexactPartial<{
       /** The human-readable (localized) label for the animation */
-      label?: string;
+      label: string;
 
       /** The animation function that runs every frame */
-      animation?: (dt: number, options: Partial<RenderedPointSourceAnimationConfig>) => any;
+      animation: (
+        dt: number,
+        options: InexactPartial<RenderedPointSourceAnimationConfig<AnimationValue>>,
+      ) => AnimationValue;
```

Comment:
I checked the Foundry source. I suggested generics if it could be be any value but it looks like Foundry source has this to say:
```js
  /**
   * Animate the PointSource, if an animation is enabled and if it currently has rendered containers.
   * @param {number} dt         Delta time.
   */
  animate(dt) {
    if ( !this.isAnimated ) return;
    const {animation, ...options} = this.animation;
    return animation.call(this, dt, options);
  }
  ```

It assumes that `animation.call(...)` returns `number`. Sorry to make you add the generics but please remove it and make it `number`.